startblog/views.py

```diff
@@ -54,81 +80,109 @@ def blog_update(request,pk):
+                    'token': default_token_generator.make_token(user),
+                    'protocol': 'http',
+                    }
+                    email = render_to_string(email_template_name, c)
+                    try:
+                        '''delay is used as its a celery function in tasks.py '''
+                        send_mail_task.delay(subject,email,'sender@gmail.com',user.email)
+                    except BadHeaderError:
+                        return HttpResponse('Invalid header found.')
+                    return redirect ("/password_reset/done/")
+    password_reset_form = PasswordResetForm()
+    return render(request=request, template_name="passwords/password_reset.html", context={"password_reset_form":password_reset_form})
+
 
 def add_like(request,pk):
     blog = Blogs.objects.get(pk=pk)  
     if request.method =='POST':
-        if request.POST.get('like')=='like':
-            like =Likes.objects.filter(post=blog,like_status=False,liked_user=request.user)   
+        if request.POST.get('like') ==' like':
+            like =Likes.objects.filter(post=blog, like_status=False, liked_user=request.user)   
             if like:
                 like.update(like_status = True)
             else:
-                like = Likes.objects.create(post=blog,like_status=True,liked_user=request.user)
+                like = Likes.objects.create(post=blog, like_status=True, liked_user=request.user)
                 return redirect('startblog:home')
-        if request.POST.get('like')=='unlike':
+        if request.POST.get('like') == 'unlike':
             Likes.objects.get(post=blog,liked_user=request.user).delete()
-    return redirect('startblog:home') 
+    return redirect('startblog:home')
 
-def add_comment(request,pk):
-    if request.method=='POST':
+
+def add_comment(request, pk):
+    if request.method == 'POST':
         cmnt = request.POST.get('comment')
         blog = Blogs.objects.get(pk=pk)
-        coment = Comments.objects.create(post=blog,cmnt=cmnt,comented_by = request.user)
+        coment = Comments.objects.create(post=blog, cmnt=cmnt, comented_by = request.user)
         coment.save();
+        return redirect('startblog:home')
     return redirect('startblog:home')    
-   
-def view_comments(request,pk):
+
+
+def view_comments(request, pk):
     blog = Blogs.objects.get(pk=pk)
-    print(blog)
     coment_list = Comments.objects.filter(post_id=pk)  
-    context = {'coment_list':coment_list,'blog':blog}
-    return render(request,'startblog/view-comments.html',context)
\ No newline at end of file
+    context = {'coment_list': coment_list,'blog': blog}
+    return render(request,'startblog/view-comments.html', context)
+
+
+def comment_delete(request,pk):
+    if request.method == 'POST':
+        com = Comments.objects.get(pk=pk)
+        com.delete()
+        return redirect('startblog:home')
+
+
+def bloger_profile(request, pk):
+    user = User.objects.get(pk=pk)
+    if request.method == 'POST':
```

Comment:
Use DetailView and UpdateView for viewing and updating a user

Reply:
Done